### PR TITLE
chore(sdk): move all code checks to use noxfile

### PR DESCRIPTION
**Pull request by kptkin**
Description
-----------

What does the PR do? Include a concise description of the PR contents.

- [x] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -281,7 +280,7 @@ ensure that URLs do not lead to broken links.
 
 Once you add the URL to that file you will need to run:
 ```shell
-tox -e auto-codegen
+nox -s codegens
```
Comment (timoffex) [inline CONTRIBUTING.md:283]:
```suggestion
nox -s codegen
```

File: noxfile.py [hunk 2]
```diff
@@ -336,3 +315,59 @@ def proto_check_go(session: nox.Session) -> None:
...
+
+@nox.session(name="mypy-report")
+def mypy(session: nox.Session) -> None:
+    """Type-check the code with mypy.
+
+    This session will install the package and run mypy with the --install-types flag.
+    If the report parameter is set to True, it will also generate an html report.
+    """
+    session.install("mypy")
+    session.install(".")
```
Comment (timoffex) [inline noxfile.py:339]: why do we have to install `.` here?

File: wandb/sdk/launch/registry/google_artifact_registry.py [hunk 3]
```diff
@@ -211,7 +211,7 @@ async def check_image_exists(self, image_uri: str) -> bool:
             for image in await list_images(request={"parent": parent}):
                 if tag in image.tags:
                     return True
-        except google.api_core.exceptions.NotFound as e:
+        except google.api_core.exceptions.NotFound as e:  # type: ignore[attr-defined]
```
Comment (timoffex) [inline wandb/sdk/launch/registry/google_artifact_registry.py:214]: what's the issue here?

**Review by timoffex — APPROVED** [review]

File: noxfile.py (hunk 2 above)
Comment (kptkin) [reply noxfile.py:339]: Ah good point I don’t think we do

File: wandb/sdk/launch/registry/google_artifact_registry.py (hunk 3 above)
Comment (kptkin) [reply wandb/sdk/launch/registry/google_artifact_registry.py:214]: missing Google types

Comment (kptkin) [reply wandb/sdk/launch/registry/google_artifact_registry.py:214]: I’ll look on type shad or something, but cursory search didn’t yield much

File: .circleci/config.yml [hunk 4]
```diff
@@ -581,10 +581,13 @@ jobs:
...
-          command: nox -t proto-check
+          command: nox -vt proto-check
       - run:
-          name: Lint
-          command: tox run -v -e "auto-codegen-check, mypy, mypy-report"
+          name: Automatically generate code and check for changes
+          command: nox -vs codegen -- --check
+      - run:
+          name: Run mypy and report results
+          command: nox -s mypy-report
```
Comment (kptkin) [inline .circleci/config.yml:590]:
```suggestion
          command: nox -vs mypy-report
```

File: noxfile.py [hunk 5]
```diff
@@ -336,3 +315,59 @@ def proto_check_go(session: nox.Session) -> None:
...
+
+    args = session.posargs
+    if not args:
+        args = ["--generate"]
+    session.run("python", "tools/generate-tool.py", *args)
+
+
+@nox.session(name="mypy-report")
+def mypy(session: nox.Session) -> None:
```
Comment (kptkin) [inline noxfile.py:332]:
```suggestion
def mypy_report(session: nox.Session) -> None:
```

**Pull request merged**